### PR TITLE
Club Base Components

### PR DESCRIPTION
**Pull request by 7alip**
> This implementation is based on server side data fetching. If we prefer to implement filtering and pagination on the client side, you can take it as a task, investigate and and create a new pull request.

Example link: https://wsvvrijheid-git-feat-club-page-samenvvv.vercel.app/tr/club?page=1&category=photo

###  Why `getServerSideProps` instead of `getStaticProps`?

According to one of the maintainers of Nextjs this is the reply for anyone learning this framework:

> getStaticProps generates the page at build time. There's no possible way to know the custom query params your visitors can use at build time.

> getInitialProps and getServerSideProps can know all the possible query params because the page is generated at runtime, whenever you receive a new request.

[Source](https://stackoverflow.com/a/64159389/8206907)

### What's next?

This request was necessary to clear up the confusion about data fetching strategy. **<ins>After merging this request<ins>**, you can continue adding/improving your features (#27  @dnlrkaan, #29 @yuskara and #28 @yhtyyar94) based on the new develop branch.

### CODE REVIEW

File: src/hooks/useChangeParams.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    const [param, value] = Object.entries(args)[0]
+
+    let { page } = router.query
+
+    // If the params is not page, then we set it to 1 by default
+    if (param !== 'page') page = 1
+
+    return router.push({
+      // Finally we set the query with the new params
+      query: { ...router.query, page, [param]: value },
```
Comment (yuskara) [inline src/hooks/useChangeParams.js:23]: does the page line necessary here ? as it is already in router.query

File: src/hooks/useChangeParams.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  // Example
+  // args { category: "photo", page: 4, sort: "likes" }
+  // query { category: "photo", page: "4", sort: "likes" }
+  // `/?category=photo&page=4&sort=likes`
+  const changeParam = args => {
+    // Extract the first key-value pair from the object
+    // Example { page: 1 } => [ 'page', 1 ]
+    // Example { category: "photo" } => [ 'category', 'photo' ]
+    const [param, value] = Object.entries(args)[0]
```
Comment (yuskara) [inline src/hooks/useChangeParams.js:14]: what if we provide multiple params at the same time?

Comment (7alip) [reply src/hooks/useChangeParams.js]: We probably won't need to provide multiple params at the same time but for the complex filtering/sorting scenarios we may need that. Thanks for the warning, I updated the function with a better structure.

**Comment by yuskara** [comment]
thank you for your clear code

**Pull request merged**